test/ChessPiecesTest.hs

```diff
@@ -21,6 +21,16 @@ test_getPieceColour =
         assertEqual Black (getPieceColour (Pawn Black))
 
 
+test_legalNextPosForPieceAtPos_King = 
```

Comment:
I think we need more exhaustive tests here. The tests should cover at least the cases where:
* The king is at the various edges of the board (both colours)
* The king is faced by an enemy piece (both colours, on the left/right, on the diagonal)
* The king is faced by a friendly piece (both colours, on the left/right, on the diagonal)

You can cover most of those cases in just one chess board (it doesn't have to be a valid chess board). Here's an example: https://lichess.org/editor/1rkK3k/2Q3Pn/K7/N4Np1/6K1/5r1n/k1q5/B5K1_w_-_-_0_1

Reply:
PS: to convert that link into the `ChessBoard` type just copy the part of the link after `editor/` until the underscore `_w...`. So in this case it would be `fenToChessBoard "1rkK3k/2Q3Pn/K7/N4Np1/6K1/5r1n/k1q5/B5K1"`